src/gen_qr_lib.js

```diff
@@ -32,6 +32,19 @@ function keccak128 (input) {
     return keccak256(input).slice(0, 16);
 }
 
+function ensureDirectoryExistence(dir) {
+    // Ensure the directory exists, create it recursively if not
+    if (!fs.existsSync(dir)) {
+        fs.mkdirSync(dir, { recursive: true });
+    }
+}
+
+function saveFile(filePath, fileContent) {
+    const dir = path.dirname(filePath);
+    ensureDirectoryExistence(dir);
```

Comment:
I don't like the idea to check the directory existence before each save having in mind that all files are saved in the same directory. It's better to check existence before saving everything.